### ISSUE TITLE
False positive of IM_MULTIPLYING_RESULT_OF_IREM

### ISSUE THREAD

**Issue by davidmoten** [spotbugs/spotbugs#1828]
Using 4.5.0 I now have to deal with a false positive of IM_MULTIPLYING_RESULT_OF_IREM for code that looks like this:

```java
a = (b % 10) * c;
```

I checked the commit 0e33efd that brought this about which was to fix #1498.

I see that spotbugs makes its decision based on compiled opcodes so would miss entirely the fact someone has taken away possible ambiguity using brackets. 

My suggestion is that the documentation for IM_MULTIPLYING_RESULT_OF_IREM be more specific about how to avoid it being flagged. Currently it [says](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#im-integer-multiply-of-result-of-integer-remainder-im-multiplying-result-of-irem):

> IM: Integer multiply of result of integer remainder (IM_MULTIPLYING_RESULT_OF_IREM)
>
>The code multiplies the result of an integer remaining by an integer constant. Be sure you don't have your operator precedence confused. For example i % 60 * 1000 is (i % 60) * 1000, not i % (60 * 1000).

In my code example there is definitely no confusion of operator precedence because it has been forced by brackets (even if not strictly needed) but because spotbugs can't detect this using opcodes I think the bug description needs to be specific about how to become compliant which is something like:

>The code multiplies the result of an integer remaining by an integer constant. Be sure you don't have your operator precedence confused. For example i % 60 * 1000 is (i % 60) * 1000, not i % (60 * 1000). *To avoid confusion assign the integer remaining to a variable first.*
